app/models/manageiq/providers/vmware/cloud_manager/orchestration_template.rb

```diff
@@ -29,7 +31,64 @@ def vapp_parameters
     ]
   end
 
+  def vm_param_groups
+    # Parse template's OVF file
+    ovf_doc = MiqXml.load(content)
+
+    groups = []
+    # Parse the XML template document for specific vCloud attributes.
+    ovf_doc.root.each_element("//vcloud:GuestCustomizationSection") do |el|
+      vm_id = el.elements["vcloud:VirtualMachineId"].text
+      vm_name = el.elements["vcloud:ComputerName"].text
+
+      groups << OrchestrationTemplate::OrchestrationParameterGroup.new(
+        :label      => vm_name,
+        :parameters => [
+          # Name of the provisioned instance.
+          OrchestrationTemplate::OrchestrationParameter.new(
+            :name          => "instance_name-#{vm_id}",
+            :label         => "Instance name",
+            :data_type     => "string",
+            :default_value => vm_name
+          ),
+
+          # List of available VDC networks.
+          OrchestrationTemplate::OrchestrationParameter.new(
+            :name          => "vdc_network-#{vm_id}",
+            :label         => "Network",
+            :data_type     => "string",
+            :default_value => "(default)",
+            :constraints   => [
+              OrchestrationTemplate::OrchestrationParameterAllowed.new(:allowed_values => vdc_networks)
+            ]
+          )
+        ]
+      )
+    end
+
+    groups
+  end
+
   def self.eligible_manager_types
     [ManageIQ::Providers::Vmware::CloudManager]
   end
+
+  private
+
+  def vdc_networks
+    # In order to be able to deploy this vApp we need to have a valid EMS.
+    ems = ExtManagementSystem.find(ems_id) unless ems_id.nil?
+
+    # Get all VDC networks from the provider
+    vdc_network_type = ManageIQ::Providers::Vmware::NetworkManager::CloudNetwork::OrgVdcNet.name
+    cloud_networks = ems.try(:cloud_networks).try(:where, :type => vdc_network_type)
+
+    # If none exists, ensure user can't select anything (default network will be used)
+    return "(default)" if cloud_networks.empty?
+
+    # Get network names.
+    vdc_network_names = cloud_networks.collect(&:name)
```

Comment:
Here I am only collecting names of the networks, however I'd rather see if I could create `key,value` pairs. The list of names is used [here](https://github.com/ManageIQ/manageiq/pull/12273/files#diff-d27d263ac8ea2329188a0a111567b5e5R62) and converted into the actual dialog field in `OrchestrationTemplateDialogService` [class](https://github.com/ManageIQ/manageiq/blob/9ed3cad31b10f326515d388071799e59e0e54518/app/services/orchestration_template_dialog_service.rb#L216).

What I am thinking is if it would be possible to change the code so that it would allow setting up `key, value` and the corresponding code would use keys if they exist, otherwise it would use the same logic as today.